### PR TITLE
odometry: add node to compute walk odometry

### PR DESCRIPTION
**Pull request by SammyRamone**
this adds a node which computes the odometry of the walking based on the actual joint states rather than the commands send by the walking. It should therefore be more accurate
should solve https://github.com/bit-bots/bitbots_motion/issues/54
Currently not tested, therefore a draft

- [x] adapt odometry fuser
- [x] adapt walk publishing of support foot

## Necessary checks
- [x] Update package version
- [x] Run linters
- [x] Run `catkin build`
- [x] Write documentation
- [x] Test on your machine
- [x] Test on the robot

### CODE REVIEW

**Comment by SammyRamone** [comment]
tested in visualization, but not on robot yet

**Comment by timonegk** [comment]
The node would publish odometry for other motions like kick or standing up too

**Comment by NFiedler** [comment]
@timonegk that should be tested, but listening on /cmd_vel could resolve that in case it leads to errors.

File: bitbots_quintic_walk/config/walking_wolfgang_viz.yaml [hunk 1]
```diff
@@ -71,7 +71,7 @@ walking:
 
   node:
     # update frequency of the engine
-    engine_freq: 200.0
+    engine_freq: 100.0
```
[the remark this answers is not in the document]
Comment (SammyRamone) [reply bitbots_quintic_walk/config/walking_wolfgang_viz.yaml]: yes, these are the visualization parameters. 100hz is enough for that and does not stress your laptop as much

[the remark this answers is not in the document]
Comment (SammyRamone) [reply bitbots_quintic_walk/config/walking_wolfgang_viz.yaml]: could have been put in a seperate PR, but I thought that its small enough to smuggle it in here :-P

File: bitbots_odometry/src/walk_odometry.cpp [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        tf2::Transform odom_to_base_link = odometry_to_support_foot * current_support_to_base;
+        geometry_msgs::TransformStamped odom_to_base_link_msg = geometry_msgs::TransformStamped();
+        odom_to_base_link_msg.transform = tf2::toMsg(odom_to_base_link);
+        odom_to_base_link_msg.header.stamp = ros::Time::now();
+        odom_to_base_link_msg.header.frame_id = "odom";
+        odom_to_base_link_msg.child_frame_id = "base_link";
+        br.sendTransform(odom_to_base_link_msg);
+      } catch (tf2::TransformException &ex) {
+        ROS_WARN("%s", ex.what());
+        ros::Duration(1.0).sleep();
```
[the remark this answers is not in the document]
Comment (SammyRamone) [reply bitbots_odometry/src/walk_odometry.cpp]: changed it

File: bitbots_quintic_walk/launch/quintic_walk.launch [hunk 3]
```diff
@@ -17,6 +17,7 @@
   </group>   
     
   <node name="walking" pkg="bitbots_quintic_walk" type="WalkNode" respawn="true" output="screen"/>
+  <node name="walk_odometry" pkg="bitbots_odometry" type="walk_odometry" respawn="true" output="screen"/>
```
[the remark this answers is not in the document]
Comment (SammyRamone) [reply bitbots_quintic_walk/launch/quintic_walk.launch]: added it as test dependency

File: bitbots_odometry/src/walk_odometry.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (timonegk) [reply bitbots_odometry/src/walk_odometry.cpp]: Same thing in line 72

**Comment by NFiedler** [comment]
We need to adapt the odometry fuser.

**Comment by NFiedler** [comment]
Currently, there is an issue with short double support phases (if no 'd' message is published). @SammyRamone , could we solve this by reacting on changes from 'r' to 'l' (and backwards respectively?

**Comment by SammyRamone** [comment]
> Currently, there is an issue with short double support phases (if no 'd' message is published). @SammyRamone , could we solve this by reacting on changes from 'r' to 'l' (and backwards respectively?

I will rewrite the walk, so that it will only publish the support foot when changing it

**Comment by SammyRamone** [comment]
integrated it with the odometry fuser and tested on my machine

**Comment by SammyRamone** [comment]
tested on robot works, can be merged

**Pull request merged**